test/python/test_roles_api.py

```diff
@@ -4,24 +4,714 @@
+
+    def test_get_members_422(self):
+        """Test case for 422 status response for GET requests using 'members' query parameter
+        """
+        with self.assertRaises(openapi_client.ApiException) as context:
+            self.api.get_role_with_http_info(
+                self.account,
+                'user',
+                'admin',
+                members='',
+                search=NULL_BYTE
+            )
+
+        self.assertEqual(context.exception.status, 422)
+
+    # Test cases confirming count, offset, limit and search query parameters
+
+    def test_get_members_count(self):
+        """Test using members & count query parameters on GET
+        Adds dev:user:bob as a member of group:userGroup, then queries for a member count
+        """
+        self.add_user_to_group("bob")
+        self.add_user_to_group("alice")
+
+        response = self.api.get_role(
+            self.account,
+            'group',
+            'userGroup',
+            members='',
+            count=True
+        )
+
+        self.assertEqual(response['count'], 3)
+
+    def test_get_members_offset(self):
+        """Test using members & offset query parameters on GET requests
+        Returns member data starting at <offset>
+        """
+        self.add_user_to_group("bob")
+        self.add_user_to_group("alice")
+
+        response = self.api.get_role(
+            self.account,
+            'group',
+            'userGroup',
+            members='',
+            offset=2
+        )
+
+        # starting at offset 2 should omit user:admin (0) and user:alice (1)
+        target_response = [
+            {
+                'admin_option': False,
+                'ownership': False,
+                'role': self.GROUP_ID,
+                'member': self.BOB_ID
+            }
+        ]
+
+        self.assertEqual(response, target_response)
+
+    def test_get_members_limit(self):
+        """Test using members & limit query parameters on GET requests
+        Returns member data limited to the first <limit> results
+        """
+        self.add_user_to_group("bob")
+        self.add_user_to_group("alice")
+
+        response = self.api.get_role(
+            self.account,
+            'group',
+            'userGroup',
+            members='',
+            limit=2
+        )
+
+        # limiting the results to the first 2 should omit user:bob (idx 2)
+        target_response = [
+            {
+                'admin_option': True,
+                'ownership': True,
+                'role': self.GROUP_ID,
+                'member': self.ADMIN_ID,
+                'policy': self.ROOT_ID
+            },
+            {
+                'admin_option': False,
+                'ownership': False,
+                'role': self.GROUP_ID,
+                'member': self.ALICE_ID
+            }
+        ]
+
+        self.assertEqual(response, target_response)
+
+    def test_get_members_search(self):
```

Comment:
Did you test each combination of each optional query parameter together? Or do you know if certain parameters are not supposed to work together? I think it would be out of scope for these tests, but it seems like an interesting question.

Reply:
I haven't. Seems similar to a conversation @izgeri and I were having about query parameters on Resources endpoints. I filed an [issue](https://github.com/cyberark/conjur-openapi-spec/issues/114) for testing and documenting these relationships on both Roles and Resources endpoints.